loan-payment-skip/REPWRITERSPECS/BANNO.LOANPAYMENT.SKIP.V1.POW

```diff
@@ -1063,5 +1674,110 @@ PROCEDURE CLEANUPTEXT
  TMPCHR=TMPTEXT
 END [PROCEDURE]
 
+PROCEDURE BNOPRINTPROGRAMINFO
+[* Print program info to direct JSON output
+**
+**  PASS    BNOPROGRAMNAME     Program name
+**  PASS    BNOPROGRAMVERSION  Program version
+**  PASS    BNOLASTMODDATE     Last mod date
+**  PASS    BNOLASTMODTIME     Last mod time
+*]
+ PRINT BNOINDENT(1)+","
+ NEWLINE
+ PRINT BNOINDENT(1)+Q+"programInfo"+Q+":"+Q+
+                    BNOPROGRAMNAME+"  "+
+                    BNOPROGRAMVERSION+"  "+
+                    FORMAT("99/99/99 ",BNOLASTMODDATE)+"  "+
+                    "TESTMODE:"+TRUEFALSECHR(TESTMODE)+"  "+
+                    "TESTACCT:"+TRUEFALSECHR(VALIDTESTACCOUNT)+Q
+END [PROCEDURE]
+
+PROCEDURE BNOPRINTRGDATA
+[* Print @RGUSER[CHR,NUM] data passed from the UX in proper JSON format
+**
+**  PASS    DEBUGMODE        Boolean: Print debug data, T/F
+*]
+
+ IF BNODEBUGMODE=BNODEBUGMODEON THEN
+  DO
+
+   PRINT ","
+   NEWLINE
+   PRINT BNOINDENT(1)+Q+"rgData"+Q+": ["
+   NEWLINE
+   PRINT BNOINDENT(2)+Q+"CHR1:'"+@RGUSERCHR1+"' "+
+                     "CHR2:'"+@RGUSERCHR2+"'"+Q+","
+   NEWLINE
+   PRINT BNOINDENT(2)+Q+"CHR3:'"+@RGUSERCHR3+"' "+
+                     "CHR4:'"+@RGUSERCHR4+"'"+Q+","
+
+   NEWLINE
+   PRINT BNOINDENT(2)+Q+"CHR5:'"+@RGUSERCHR5+"' "+
+                     "NUM1:"+FORMAT("#########9 ",@RGUSERNUM1)+
+                     "NUM2:"+FORMAT("#########9 ",@RGUSERNUM2)+
+                     "NUM3:"+FORMAT("#########9 ",@RGUSERNUM3)+
+                     "NUM4:"+FORMAT("#########9 ",@RGUSERNUM4)+
+                     "NUM5:"+FORMAT("#########9",@RGUSERNUM5)+Q
+   NEWLINE
+   PRINT BNOINDENT(1)+"]"
+   NEWLINE
+  END
+END [PROCEDURE]
+
+PROCEDURE BNODEBUGMSGADD
+[* Add a message line to the debug message array
+**
+** PASS    BNODEBUGMSGMAX     Current array index max lines allowed
+** PASS    BNODEBUGMSG        Debug message to add to array
+**
+** RETURN  BNODEBUGCOUNT      Count of debug message lines
+** RETURN  BNODEBUGMSGLIST(A) Debug lines of data to print
+*]
+ IF BNODEBUGCOUNT<BNODEBUGMSGMAX THEN
+  DO
+   BNODEBUGCOUNT=BNODEBUGCOUNT+1
+   BNODEBUGMSGLIST(BNODEBUGCOUNT)=BNODEBUGMSG
+   BNODEBUGMSG=""
+  END
+END [PROCEDURE]
+
+PROCEDURE BNOPRINTDEBUG
+[* Print debug output in proper JSON format. Requires DEBUGMODE to be
+** on (or true).
+**
+**  PASS  DEBUGMODE           Boolean: Print debug data, T/F
+**  PASS  BNODEBUGMSGLIST(A)  Array of debug messages to print
+**  PASS  BNODEBUGCOUNT       Number of debug lines to print
+*]
+ IF BNODEBUGMODE=BNODEBUGMODEON THEN
+  DO
+   PRINT ","
+   NEWLINE
+   PRINT BNOINDENT(2)+Q+"debugData"+Q+": ["
+   NEWLINE
+   PRINT BNOINDENT(2)+Q+"Debug note: Debug lines truncated to 126 characters"+Q
+
+   FOR BNODEBUGLOOP=1 TO BNODEBUGCOUNT
+    DO
+     IF BNODEBUGLOOP=1 THEN
+      DO
+       IF BNODEBUGLOOP<BNODEBUGCOUNT THEN
+        DO
+         PRINT ","
+         NEWLINE
+        END
+      END
+     PRINT BNOINDENT(3)+Q+Segment(BNODEBUGMSGLIST(BNODEBUGLOOP),1,126)+Q

```

Comment:
On line 1770, since you are limiting the messages to 126 characters, does it matter that in the condition that BNODEBUGCOUNT is 1, that there is now NEWLINE before Printing the BNODEBUGMSGLIS? Possibly doesn't mater or would never happened. Was comparing this to the same procedure in M2M Transfers